### PR TITLE
add yaml validation

### PR DESCRIPTION
**Pull request by nr-kkenney**
## Summary

* added new workflow which runs yamllint on push
* added yamllint config file

## Example Output
Screen Shot 2022-04-14 at 9 57 54 AM

### CODE REVIEW

File: .github/workflows/yaml-lint.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Lint Yaml
+
+on:
+  push:
+    paths:
+      - '**/*.yml'
```
Comment (nr-kkenney) [inline .github/workflows/yaml-lint.yml:6]: probably also want .yaml

Comment (nr-kkenney) [reply .github/workflows/yaml-lint.yml:6]: also looks like the pattern can just be **.yml, **.yaml

File: .github/workflows/yaml-lint.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: Lint Yaml
+
+on:
+  push:
```
Comment (nr-kkenney) [inline .github/workflows/yaml-lint.yml:4]: should be `pull_request` i think, if a commit doesn't have yml files but the PR in general does.

File: .config/yamllint/config.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+
+extends: relaxed
+
+ignore: |
+  **/node_modules/
```
Comment (nr-kkenney) [inline .config/yamllint/config.yml:6]: might want to also exclude testing folder mock_files

File: .github/workflows/yaml-lint.yml (hunk 2 above)
Comment (nr-kkenney) [reply .github/workflows/yaml-lint.yml:4]: pull_request should also work with forks, since we don't need secrets or write permissions to run validation

**Review by aswanson-nr — APPROVED** [review]

**Pull request merged**